workers/common/src/test/kotlin/common/env/MavenSettingsGeneratorTest.kt

```diff
@@ -110,6 +120,136 @@ class MavenSettingsGeneratorTest : WordSpec({
             lines[lines.size - 2] shouldBe "    </servers>"
             lines[lines.size - 1] shouldBe "</settings>"
         }
+
+        "generates no proxy section if system properties are not set" {
+            mockkObject(HTTP_PROXY_HOST)
```

Comment:
KoTest has a [helper function](https://kotest.io/docs/extensions/system_extensions.html#system-property-extension) to temporarily override system properties. Using this is probably simpler than mocking the proxy constants. If you use mocking, please place the `unmockAll()` call in an `afterTest` function to make sure that it is always invoked, even in case of a test failure.

Reply:
Thanks for the hint. That makes the test look a lot of nicer.